packages/polling-controller/src/PollingController.ts

```diff
@@ -38,24 +41,53 @@ function PollingControllerMixin<TBase extends Constructor>(Base: TBase) {
 
     readonly #intervalIds: Record<PollingTokenSetId, NodeJS.Timeout> = {};
 
+    readonly #activeListeners: Record<
+      PollingTokenSetId,
+      (options: Json) => Promise<void>
+    > = {};
+
     #callbacks: Map<
       NetworkClientId,
       Set<(networkClientId: NetworkClientId) => void>
     > = new Map();
 
-    #intervalLength = 1000;
+    #intervalLength: number | undefined = 1000;
+
+    #getNetworkClientById:
+      | ((networkClientId: NetworkClientId) => NetworkClient)
+      | undefined;
 
     getIntervalLength() {
       return this.#intervalLength;
     }
 
+    getPollingWithBlockTracker() {
+      return this.#getNetworkClientById !== undefined;
+    }
+
     /**
      * Sets the length of the polling interval
      *
      * @param length - The length of the polling interval in milliseconds
      */
     setIntervalLength(length: number) {
       this.#intervalLength = length;
+
+      // setting and using an interval is mutually exclusive with polling on new blocks
+      this.#getNetworkClientById = undefined;
+    }
+
+    setPollWithBlockTracker(
+      getNetworkClientById: (networkClientId: NetworkClientId) => NetworkClient,
```

Comment:
I wasn't sure if this argument would be best as a `ControllerMessenger` as this `(Network)PollingController` can know how to get a network client given an ID, plus it encourages messenger adoption.

But I suppose the counter-argument is that it requires all child controllers to `BaseController2` or have a messenger.